### PR TITLE
Standalone kubeconfig for gce kube-up (mulligan)

### PR DESCRIPTION
**Pull request by j3ffml**
Un-revert of #5154. Added check to create empty `.kubeconfig` if the provided one doesn't exist. That should make it work on jenkins, which runs e2e from a redirected, empty $HOME.

cc @zmerlynn, @satnam6502

### CODE REVIEW

File: cluster/common.sh [hunk 1]
```diff
@@ -0,0 +1,75 @@
...
+#   KUBE_KEY
+#   CA_CERT
+#   CONTEXT
+function create-kubeconfig() {
+  local kubectl="${KUBE_ROOT}/cluster/kubectl.sh"
+
+  # KUBECONFIG determines the file we write to, but it may not exist yet
+  if [[ ! -e "${KUBECONFIG}" ]]; then
+    mkdir -p $(dirname "${KUBECONFIG}")
+    touch "${KUBECONFIG}"
```
Comment (zmerlynn) [inline cluster/common.sh]: Do you care about the umask here?

Comment (jlowdermilk) [inline cluster/common.sh]: I think 077 is correct. Or do you mean I should explicitly set umask here?

Comment (zmerlynn) [inline cluster/common.sh]: Yeah, I was more asking if we should be explicit here. I don't think it matters much, though.

Comment (zmerlynn) [inline cluster/common.sh]: ... we're not explicit elsewhere. So. Let's proceed for now.

Comment (jlowdermilk) [inline cluster/common.sh]: It doesn't actually matter. We'll be using `kubectl` to write to the file in the next line, and `kubectl` sets permissions [explicitly](https://github.com/GoogleCloudPlatform/kubernetes/blob/master/pkg/client/clientcmd/loader.go#L205).

**Pull request merged**